### DISCUSSION TITLE
Icon for appimage application in the Ubuntu bar

### DISCUSSION THREAD

**Discussion by HDInfautre** [keepassxreboot/keepassxc#10978] [Q&A] [answered]
I have several images with the same default Ubuntu icon.
Is it possible to add the keepass icon?, see a name different from keepassXC?
![Capture d’écran du 2024-06-25 22-20-50](https://github.com/keepassxreboot/keepassxc/assets/6817936/c1a605b5-f18d-4128-885d-3305d2a1d5ba)

**Comment by phoerious** [comment, marked as answer]
AppImages have no application icon. You have to install them using AppImageKit if you want a launcher. Or write your own .desktop file.

**Reply by michaelk83** [reply]
On KDE, you can right-click the icon on the app bar and select "Properties". You can change which icon is used from there. There is likely a similar option in Gnome and other DEs.

**Comment by Samueru-sama** [comment]
The issue is gnome, a recent discussion popped about this on [reddit](https://www.reddit.com/r/AppImage/comments/1dhyuif/why_would_an_api_be_needed_to_display_appimage/)

**Comment by HDInfautre** [comment]
Hi
Thank you for following.
I understood in these discussions that there is a problem but I don't know how to apply a solution to the tire
I tried without success
https://acreom.com/docs/add-appimage-icon
https://snippets.page/posts/how-to-create-an-applications-menu-icon-for-appimages-in-ubuntu/

I will need a tutorial like I shared which allows on ubuntu 24.04+ to declare in the configuration of my portable appImage in the folder "/AppImageName.AppImage.home"
Can you explain this solution to me or give me a tutorial because I'm not very good at Linux

**Reply by Samueru-sama** [reply]
The first tutorial is how it is done, although you would create this [.desktop](https://raw.githubusercontent.com/keepassxreboot/keepassxc/develop/share/linux/org.keepassxc.KeePassXC.desktop.in) instead and edit the `Exec=` `TryExec=` and `Icon=` for the path to the AppImage and the path to the icon you want to show.

But also, I noticed you typed `AppImageName.AppImage.home` something aint right there. If you want an AppImage to have a portable `$HOME` you place a directory next to the appimage with the name of the appimage + `.home` and that's a whole different thing from the issue you have I think. 

>Can you explain this solution to me or give me a tutorial because I'm not very good at Linux

This is just a suggestion, but look at something like linux mint with kde if you are new to linux.

**Comment by HDInfautre** [comment]
Hi
Thank you for the quick reply. I will try again.
I've been using Ubuntu for over 10 years but this is the type of problem where I find myself limited.
I've been using appImage for years without any problem by moving my "portable disk" to another computer.
I was wondering if my architecture could pose a problem in this use case.
My current structure, on my portable disk is for example:
```
/media/rv/data/Com. computer/Appli-data-port/Bauh/
/bauh.AppImage.home/
bauh.AppImage
```
Is there a better way to name this directory?

**Reply by Samueru-sama** [reply]
No, in that case that's ok. 

You would want to edit the `.desktop` to point to the path where you have the appimage.

There is also this tool for [appimages](https://github.com/ivan-hc/AM) which does a lot of the things you are doing automatically, including creating the `.desktop`, read the description a bit as it details everything that can be done.

**Reply by HDInfautre** [reply]
thank you for the answer
I'm looking at this and especially the application which looks very good but it will take me a little time

**Comment by HDInfautre** [comment]
HI
In the end I succeeded with this superb application, which is full of promise https://github.com/mijorus/gearlever because I didn't manage with the other solutions.
